aptos-move/block-executor/src/executor.rs

```diff
@@ -263,6 +229,11 @@ where
             scheduler.finish_abort(idx_to_validate, incarnation)
         } else {
             scheduler.finish_validation(idx_to_validate, validation_wave);
+
+            if valid {
```

Comment:
I'd love if we can come up with another precondition for when to arm (e.g. validated index happens to be the next one that needs to be committed). If we do that, we may even be able to just use that as an indicator to do the coordinating duty right after (without extra logic for it there)